src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py

```diff
@@ -14,38 +14,74 @@
+        return permission_class(_str=string)
+    return validator
+
+def table_permission_validator(string):
+    """ A special case for table because the SDK associates the QUERY permission with 'r' """
     if set(string) - set('raud'):
-        raise ValueError('valid values are (r)ead, (a)dd, (u)pdate, (d)elete or a combination ' + \
-                         'thereof (ex: ra)')
-    return TablePermissions(_str=''.join(set(string)))
-
-def validate_queue_permission(string):
-    ''' Validates that permission string contains only a combination
-    of (r)ead, (a)dd, (u)pdate, (p)rocess [delete]. '''
-    if set(string) - set('raup'):
-        raise ValueError('valid values are (r)ead, (a)dd, (u)pdate, (p)rocess [delete] or a '
-                         'combination thereof (ex: ra)')
-    return QueuePermissions(_str=''.join(set(string)))
+        help_string = '(r)ead/query (a)dd (u)pdate (d)elete'
+        raise ValueError('valid values are {} or a combination thereof.'.format(help_string))
+    return TablePermissions(_str=string)
+
+def get_content_setting_validator(settings_class):
+    def validator(namespace):
+        namespace.content_settings = settings_class(
+            content_type=namespace.content_type,
+            content_disposition=namespace.content_disposition,
+            content_encoding=namespace.content_encoding,
+            content_language=namespace.content_language,
+            content_md5=namespace.content_md5,
+            cache_control=namespace.content_cache_control
+        )
+        del namespace.content_type,
+        del namespace.content_disposition,
+        del namespace.content_encoding,
+        del namespace.content_language,
+        del namespace.content_md5,
+        del namespace.content_cache_control
+    return validator
+
+def process_logging_update_namespace(namespace):
+    services = namespace.services
+    if set(services) - set('bqt'):
+        raise ValueError('--services: valid values are (b)lob (q)ueue '
+                         '(t)able or a combination thereof (ex: bt).')
+    log = namespace.log
+    if set(log) - set('rwd'):
+        raise ValueError('--log: valid values are (r)ead (w)rite (d)elete '
+                         'or a combination thereof (ex: rw).')
+
+def process_metric_update_namespace(namespace):
+    namespace.hour = namespace.hour == 'enable'
+    namespace.minute = namespace.minute == 'enable'
+    namespace.api = namespace.api == 'enable' if namespace.api else None
+    if namespace.hour is None and namespace.minute is None:
+        raise argparse.ArgumentError(
+            None, 'incorrect usage: must specify --hour and/or --minute')
+    if (namespace.hour or namespace.minute) and namespace.api is None:
+        raise argparse.ArgumentError(
+            None, 'incorrect usage: specify --api when hour or minute metrics are enabled')
 
 def validate_datetime_as_string(string):
```

Comment:
>validate_datetime_as_string(string) [](start = 4, length = 35)

Rename types to indicate they are types. As is they can easily be confused with validators.